pyaerocom/units/datetime/tstype.py

```diff
@@ -385,33 +393,96 @@ def _from_pandas(self, val):
             raise TemporalResolutionError(f"Invalid input: {val}, need pandas frequency string")
         return self.FROM_PANDAS[val]
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
         if isinstance(other, str):
             other = TsType(other)
         return other.val == self.val
 
-    def __lt__(self, other):
+    def __lt__(self, other) -> bool:
         if isinstance(other, str):
             other = TsType(other)
-        nss, nso = self.num_secs, other.num_secs
         # inverted comparison, i.e. if other has less seconds if has higher
         # resolution
-        return nss > nso
-
-    def __le__(self, other):
-        return True if (self.__eq__(other) or self.__lt__(other)) else False
-
-    def __gt__(self, other):
-        return not self.__le__(other)
+        return self.num_secs > other.num_secs
 
-    def __ge__(self, other):
-        return not self.__lt__(other)
-
-    def __call__(self):
-        return self.val
-
-    def __str__(self):
+    def __call__(self) -> str:
         return self.val
 
-    def __repr__(self):
-        return self.val
+    def __str__(self) -> str:
+        return str(self.val)
+
+    def __repr__(self) -> str:
+        return str(self.val)
+
+
+def sort_ts_types(ts_types: list[str | TsType]) -> list[str]:
```

Comment:
Out of scope for this PR but it would be nice if all occurrences of time series types in pyaerocom were instances of `TsType`